### DISCUSSION TITLE
An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [netblue30/firejail#5112] [Q&A]
A lot of profiles print:

```
An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
```

However, I don't see any abstract unix socket for D-Bus. In fact, it would seem that _only_ XWayland (Xorg too?) uses this:

```
$ lsof -U | grep @

sway         938 hugo   30u  unix 0x00000000408c9c67      0t0   20795 @/tmp/.X11-unix/X0 type=STREAM (LISTEN)
Xwayland    1133 hugo    4u  unix 0x0000000029f2895c      0t0   15213 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   13u  unix 0x00000000457f2c17      0t0 1884642 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   14u  unix 0x00000000ddb2ce30      0t0 1876919 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   15u  unix 0x00000000e186b997      0t0 1882867 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   16u  unix 0x00000000be8029e7      0t0 1882870 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   18u  unix 0x000000001d04d571      0t0 1882871 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   19u  unix 0x00000000f772bc33      0t0 1882872 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   20u  unix 0x00000000930060bb      0t0 1877946 @/tmp/.X11-unix/X0 type=STREAM (CONNECTED)
Xwayland    1133 hugo   30u  unix 0x00000000408c9c67      0t0   20795 @/tmp/.X11-unix/X0 type=STREAM (LISTEN)
```

I see several references online mentioning that D-Bus uses an abstract socket, but don't see it listed there -- and as far as I can tell, most applications seen to be using the filesystem-based socket.

Is `lsof -U` not the correct way to list these? Or am I missing something?

**Comment by rusty-snake** [comment]
The warning is not about an abstract D-Bus socket being present (How should you check if a socket is a D-Bus socket? What if a socket is created after the sandbox is started?). Instead it is about the possibility to access (shared) abstract sockets.

https://github.com/netblue30/firejail/blob/62e33cfc37635d985c186c8e5aaf1101070f9ccf/src/firejail/dbus.c#L533-L547

**Comment by WhyNotHugo** [comment]
Right, I understand the warning is about this _possibly_ being true, and not necessarily my case.

My question is: in what cases **is** the abstract socket used by dbus? Given that everything works without it, why would a user / distros enable it?

**Reply by rusty-snake** [reply]
> what cases is the abstract socket used by dbus? 

If it is configured to use it.
